src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -1165,7 +1204,71 @@ private enum CacheStatus {
 					FileOptions.SequentialScan);
 		}
 
+		// tries to acquire a bulk reader over a memstream but
+		// (a) doesn't block if a file reader would be acceptable instead
+		//     (we might be in the middle of caching which could take a while)
+		// (b) _does_ throw if we can't get a memstream and a filestream is not acceptable
+		private bool TryAcquireBulkMemReader(out TFChunkBulkReader reader) {
+			reader = null;
+
+			if (IsReadOnly) {
+				// chunk is definitely readonly and will remain so, so a filestream would be acceptable.
+				// we might be able to get a memstream but we don't want to wait for the lock incase we
+				// are currently performancing a slow operation with it such as caching.
```

Comment:
```suggestion
				// chunk is definitely read-only and will remain so, so a filestream would be acceptable.
				// we might be able to get a memstream but we don't want to wait for the lock in case we
				// are currently performing a slow operation with it such as caching.
```